### PR TITLE
core: rename tcb_t -> thread_t, move into thread.h

### PR DESCRIPTION
**Pull request by kaspar030**
Calling the struct "thread control block" seems outdated. This PR moves the definition into thread.h.

### CODE REVIEW

**Comment by haukepetersen** [comment]
I like!

**Comment by Kijewski** [comment]
:+1:

**Comment by DipSwitch** [comment]
1ste ACK in tha pocket

**Comment by haukepetersen** [comment]
2nd ACK (once Travis is happy)

File: core/include/thread.h [hunk 1]
```diff
@@ -22,18 +22,76 @@
...
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/**
+ * @brief Thread status list
+ * @{
+ */
+#define STATUS_NOT_FOUND        (-1)            /**< Describes an illegal thread status */
```
Comment (authmillenon) [inline core/include/thread.h]: Sorry for just writing that now, but wouldn't a `THREAD_` prefix on those be more in line with the coding conventions?

Comment (OlegHahm) [inline core/include/thread.h]: I think that's out of scope for this PR.

**Comment by kaspar030** [comment]
Well, travis did not agree at all. ;)

- factored out thread stacksize defines into "cpu_conf_common.h" for all cortex-M cpu's
- fixed some includes

@DipSwitch @haukepetersen pls take another look...

**Comment by kaspar030** [comment]
Sorry, there's more trouble.

File: core/include/thread.h [hunk 2]
```diff
@@ -22,18 +22,71 @@
...
+    char *sp;                   /**< thread's stack pointer         */
+    uint16_t status;            /**< thread's status                */
+
+    kernel_pid_t pid;           /**< thread's process id            */
+    uint16_t priority;          /**< thread's priority              */
+
+    clist_node_t rq_entry;      /**< run queue entry                */
+
+    void *wait_data;            /**< holding messages               */
+    priority_queue_t msg_waiters;   /**< threads waiting on message     */
```
Comment (DipSwitch) [inline core/include/thread.h]: Maybe the comment alignment should be fixed?

File: sys/include/pipe.h [hunk 3]
```diff
@@ -55,8 +55,8 @@ extern "C" {
 typedef struct riot_pipe
 {
     ringbuffer_t *rb;     /**< Wrapped ringbuffer. */
-    tcb_t *read_blocked;  /**< A thread that wants to write to this full pipe. */
-    tcb_t *write_blocked; /**< A thread that wants to read from this empty pipe. */
+    thread_t *read_blocked;  /**< A thread that wants to write to this full pipe. */
```
Comment (DipSwitch) [inline sys/include/pipe.h]: Comment alignment as well?

File: sys/posix/pthread/include/pthread_rwlock.h [hunk 4]
```diff
@@ -62,7 +62,7 @@ typedef struct pthread_rwlock
 typedef struct __pthread_rwlock_waiter_node
 {
```
Comment (DipSwitch) [inline sys/posix/pthread/include/pthread_rwlock.h]: maybe fix all the alignment of the comments while it's modified anyway :)

**Comment by DipSwitch** [comment]
Just a few comment alignment stuff, not that it must be fixed by me. But it would look better ;)

The code seems fine, ACK

**Comment by OlegHahm** [comment]
What's up with this PR?

**Comment by kaspar030** [comment]
I've rebased it yesterday and (hopefully) just fixed the last compile problem.

I'll unmark it WIP.
Someone should take a look at the new cpu_cortex_common.h, I had to change the include order slightly. @haukepetersen ?

**Comment by haukepetersen** [comment]
please reomve the stack size defines from the `cortexm_common/include/cpu.h`, otherwise they are twice defined for cortex MCUs...

**Comment by haukepetersen** [comment]
otherwise looks good, please squash!

**Comment by kaspar030** [comment]
- addressed, squashed

**Comment by kaspar030** [comment]
- removed riot ci dependencies

**Comment by kaspar030** [comment]
Bascally three ACKs -> go!

**Pull request merged**